### ISSUE TITLE
Special treatment of port 1?

### ISSUE THREAD

**Issue by HanabishiRecca** [arvidn/libtorrent#7315]
As you may know, port 1 has some special meaning https://github.com/arvidn/libtorrent/pull/4712.
Attempting to connect to such peers is pointless obviously. And there are lots of them generally, causing LT to make useless connection attempts.

Of course you can utilize privileged ports filter and it works:

![image](https://user-images.githubusercontent.com/13597663/218420608-b6330068-bb1e-4b59-a99c-d6b40e64844a.png)

But still, this simply generates a lot of noise. Sometimes even scaring people https://github.com/arvidn/libtorrent/issues/2284.

So maybe this will be a good idea to introduce some special treatment of port 1?
Like omit such peers and do not initiate outgoing connections to them.

**Comment by HanabishiRecca** [comment]
I guess according to https://github.com/arvidn/libtorrent/pull/4712 this is just peers using proxy.

**Comment by arvidn** [comment]
Yes, port 1 should be treated specially. I think it makes sense to just hard code ignoring it, unconditionally

**Comment by arvidn** [comment]
@HanabishiRecca does this make sense? https://github.com/arvidn/libtorrent/pull/7329

**Comment by joganol** [comment]
I do occasionally see peers connecting to me with port 1 and downloading something.
Completely blocking them is not good idea i think.

![Image](https://github.com/user-attachments/assets/de0834c1-6d50-48c9-b996-bd96adfd3578)

**Comment by HanabishiRecca** [comment]
I don't know what's the point of your comment, but sure.
